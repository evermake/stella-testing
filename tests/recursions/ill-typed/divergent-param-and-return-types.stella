language core;

extend with #fixpoint-combinator;

fn main(n : Bool) -> Bool {
  return fix(
    fn( f : fn(Bool) -> Nat) {
      return fn (a : Bool) {
        return true
      }
    }
  )(n)
}
