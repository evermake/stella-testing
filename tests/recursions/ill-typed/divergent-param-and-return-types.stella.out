An error occurred during typechecking!
Type Error Tag: [ERROR_UNEXPECTED_TYPE_FOR_EXPRESSION]
expected type
  fn (fn (Bool) -> Nat) -> fn (Bool) -> Nat
but got
  fn (fn (Bool) -> Nat) -> fn (Bool) -> Bool
for expression
  fn (f : fn (Bool) -> Nat)
{
  return fn (a : Bool)
  {
    return true
  }
}